openmc/universe.py

```diff
@@ -744,35 +748,42 @@ def bounding_region(self, bounded_type='box', boundary_type='vacuum'):
 
         check_type('boundary type', boundary_type, str)
         check_value('boundary type', boundary_type, _BOUNDARY_TYPES)
+        check_type('starting surface id', starting_id, Integral)
         check_type('bounded type', bounded_type, str)
         check_value('bounded type', bounded_type, ('box', 'sphere'))
 
-        bounding_box = self.bounding_box
+        bbox = self.bounding_box
 
         if bounded_type == 'sphere':
-            import math
-            bounding_box_center = (bounding_box[0] + bounding_box[1])/2
-            radius = math.dist(bounding_box[0], bounding_box[1])
+            bbox_center = (bbox[0] + bbox[1])/2
+            radius = np.linalg.norm(np.asarray(bbox))
             bounding_surface = openmc.Sphere(
-                x0=bounding_box_center[0],
-                y0=bounding_box_center[1],
-                z0=bounding_box_center[2],
+                surface_id=starting_id,
+                x0=bbox_center[0],
+                y0=bbox_center[1],
+                z0=bbox_center[2],
                 boundary_type=boundary_type,
                 r=radius,
             )
 
             return -bounding_surface
 
         if bounded_type == 'box':
+            surf_ids = [starting_id+i for i in range(6)]
             # defines plane surfaces for all six faces of the bounding box
-            lower_x = openmc.XPlane(bounding_box[0][0], boundary_type=boundary_type)
-            upper_x = openmc.XPlane(bounding_box[1][0], boundary_type=boundary_type)
-            lower_y = openmc.YPlane(bounding_box[0][1], boundary_type=boundary_type)
-            upper_y = openmc.YPlane(bounding_box[1][1], boundary_type=boundary_type)
-            lower_z = openmc.ZPlane(bounding_box[0][2], boundary_type=boundary_type)
-            upper_z = openmc.ZPlane(bounding_box[1][2], boundary_type=boundary_type)
+            lower_x = openmc.XPlane(bbox[0][0], surface_id=surf_ids[0])
+            upper_x = openmc.XPlane(bbox[1][0], surface_id=surf_ids[1])
+            lower_y = openmc.YPlane(bbox[0][1], surface_id=surf_ids[2])
+            upper_y = openmc.YPlane(bbox[1][1], surface_id=surf_ids[3])
+            lower_z = openmc.ZPlane(bbox[0][2], surface_id=surf_ids[4])
+            upper_z = openmc.ZPlane(bbox[1][2], surface_id=surf_ids[5])
 
-            return +lower_x & -upper_x & +lower_y & -upper_y & +lower_z & -upper_z
+            region = +lower_x & -upper_x & +lower_y & -upper_y & +lower_z & -upper_z
+
+            for surface in region.get_surfaces().values():
+                surface.boundary_type = boundary_type
```

Comment:
Just wondering if there are advantages of assigning surface boundary_type using this for surface loop instead of when constructing the surface

I would have thought
- it is quicker to assign the ``boundary_type``` when constructing the planes originally.
- With this approach we rely on ```get_surfaces``` to work for the region which seams like an extra complication and means users couldn't add extra surfaces to the region (which I guess is rare)

Reply:
I mainly did this to keep the construction of surfaces above to one line each. It felt cleaner to me. It is probably slightly slower yes, but since there's a fixed number of 6 surfaces in the region and the region expression is simple I don't think it will be a performance issue.

Reply:
Fair enough